From 23de1d0872881f525063a52e79e207ecb2727de2 Mon Sep 17 00:00:00 2001
From: Andreas Cord-Landwehr <cordlandwehr@kde.org>
Date: Sat, 2 Nov 2024 13:24:19 +0100
Subject: [PATCH] Fix reproducible build issue with ECMGeneratedHeaders

Do not generate absolute path to source header location into forwarding
header but the relative path. This behavior is the same as with AUTOMOC
generated includes.

Upstream-Status: Submitted [https://invent.kde.org/frameworks/extra-cmake-modules/-/merge_requests/485]

SPDX-FileCopyrightText: Andreas Cord-Landwehr <cordlandwehr@kde.org>
SPDX-License-Identifier: MIT
Signed-off-by: Andreas Cord-Landwehr <cordlandwehr@kde.org>
---
 modules/ECMGenerateHeaders.cmake | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/ECMGenerateHeaders.cmake b/modules/ECMGenerateHeaders.cmake
index ddd6066c..9bc5d8c5 100644
--- a/modules/ECMGenerateHeaders.cmake
+++ b/modules/ECMGenerateHeaders.cmake
@@ -196,7 +196,8 @@ function(ECM_GENERATE_HEADERS camelcase_forwarding_headers_var)
                 endif()
                 set(REGULAR_HEADER_NAME ${EGH_OUTPUT_DIR}/${originalprefix}${originalclassname}.${EGH_HEADER_EXTENSION})
                 if (NOT EXISTS ${REGULAR_HEADER_NAME})
-                    file(WRITE ${REGULAR_HEADER_NAME} "#include \"${_actualheader}\"\n")
+                    file(RELATIVE_PATH _actualheader_relative "${EGH_OUTPUT_DIR}/${originalprefix}" "${_actualheader}")
+                    file(WRITE ${REGULAR_HEADER_NAME} "#include \"${_actualheader_relative}\"\n")
                 endif()
             endif()
         endforeach()
-- 
2.43.0

